src/helpers.coffee

```diff
@@ -244,6 +248,53 @@ exports.nameWhitespaceCharacter = (string) ->
     when '\t' then 'tab'
     else string
 
+exports.locationDataToBabylon = ({first_line, first_column, last_line, last_column, range}) ->
+  loc:
+    start:
+      line: first_line + 1
+      column: first_column
+    end:
+      line: last_line + 1
+      column: last_column + 1
+  range: [
+    range[0]
+    range[1] + 1
```

Comment:
So Babel has 1-based row indexing, too? What harm would befall us if we switched to that in general, not just for AST generation?

Before we just `+ 1` after the column, we should do a test with Babel to see how it treats a `\n` at the end of a node. Is it part of the range, and causes the row to increment by 1, and then there’s a `last_column` of 1 (or zero?).

Reply:
@GeoffreyBooth ok I did a little investigating re `last_column + 1`

Looks like typically Babel doesn't include the ending `\n` in the node's location data, so usually its `loc.end.column` is on the same line. The one exception to that I was seeing is the top-level `File`/`Program` AST nodes, which apparently include the file-ending `\n` and have a `loc.end.column` of `0` on the next (nonexistent) line (after the last line of the file)

So my recommendation (in combination with the preceding comment here) is that for now we just use `last_column + 1` as I think it's unlikely to be problematic in practice

Then I think that rather than just aim to move to 1-based row indexing to align with Babel location data, we should move to using exclusive location data everywhere internally (not just ranges), as it looks like all the JS AST formats use exclusive location data. Then it seems to me that `last_column + 1` would become a non-issue as we'd be accurately tracking the row/column location data of "the character after" for everything. But let's do that in a separate scope from the AST changes